pilot/pkg/features/experimental.go

```diff
@@ -165,15 +165,23 @@ var (
 	EnableGatewayAPIGatewayClassController = env.Register("PILOT_ENABLE_GATEWAY_API_GATEWAYCLASS_CONTROLLER", true,
 		"If this is set to true, istiod will create and manage its default GatewayClasses").Get()
 
-	// EnableHBONE provides a global Pilot flag for enabling HBONE.
-	// Generally, this could be a per-proxy setting (and is, via ENABLE_HBONE node metadata).
-	// However, there are some code paths that impact all clients, hence the global flag.
-	// Warning: do not enable by default until endpoint_builder.go caching is fixed (and possibly other locations).
-	EnableHBONE = env.Register(
-		"PILOT_ENABLE_HBONE",
+	// EnableHBONEListening provides a global Pilot flag for enabling listening on HBONE.
+	EnableHBONEListening = env.Register(
+		"PILOT_ENABLE_LISTENING_HBONE",
 		false,
 		"If enabled, HBONE support can be configured for proxies. "+
 			"Note: proxies must opt in on a per-proxy basis with ENABLE_HBONE to actually get HBONE config, in addition to this flag.").Get()
+	// PreferHBONESend controls whether HBONE is preferred. If not, if an endpoint supports mTLS and HBONE, mTLS will be used
+	PreferHBONESend = env.Register(
+		"PILOT_PREFER_SENDING_HBONE",
```

Comment:
I doubt why should have such flag, mtls should always be used in this case

Reply:
There are two alternative views here:
1. HBONE is the future, will be used everywhere
2. HBONE is for interop with ambient, not desired for Envoy<->Envoy long term

If (1) is the case, then we need to prefer sending HBONE for a migration state. if we always use mTLS ,we would never migrate.

That being said, even if 1 is the _long term_ goal, its definitely not stable enough right now to make that the way to go

Reply:
Probably more than 2 views...

But also there are orthogonal things IMO. We have the auto-mtls model - of detecting what protocol the destination needs. If (1) HBONE is the future - it'll just be used in all - or most - workloads. If other protocols 
are the future - we will use them instead. 

Sidecar to sidecar in the absence of ambient is relatively stable and in no need for change.

If ztunnel OR a waypoint are enabled for a destination workload - I don't think there is any alternative except
using HBONE ( and in future if other protocols are supported - whatever protocol that destination uses ). This shouldn't need a setting since there is no other way.

What may be an option is if Envoy is running on node with ztunnel - and we want to use the sandwich on this side. Or it may do mTLS and HBONE to the destination. 

What is very clear about user intent is that the destination is ambient - either ztunnel or via waypoint, and 
it is also clear that all communication to that destination must use what is supported. Since the user has 
opted in ambient mode for this path ( client with sidecar to ambient destination ) - it is very reasonable IMO to
use the sandwich, which can greatly simplify Envoy config.